### ISSUE TITLE
Something Weird is Going on With Saved Labels

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#265]
I want to make sure that this problem is not on the backend (i.e., we have introduced a bug where we don't properly save data/labels). I just started in a new neighborhood, called Fort Lincoln and I've completed a few missions (see below).

![image](https://cloud.githubusercontent.com/assets/1621749/18029555/0eff68c6-6c69-11e6-8e96-c581be680e60.png)

However, when viewing the Fort Lincoln neighborhood in my dashboard, this auditing activity doesn't show up? :(

![image](https://cloud.githubusercontent.com/assets/1621749/18029550/f57ffafa-6c68-11e6-84ae-d4523064f625.png)

What's going on?

**Comment by jonfroehlich** [comment]
Hmm, in investigating this a bit more, looks like it might be a bug in the user profile dashboard as the admin dashboard seems to have my audit activity:

![image](https://cloud.githubusercontent.com/assets/1621749/18029569/73d82ada-6c69-11e6-9e33-a6c8d2318a07.png)

**Comment by jonfroehlich** [comment]
Yes, the user profile dashboard is definitely having problems:

![image](https://cloud.githubusercontent.com/assets/1621749/18036343/11d387ee-6d36-11e6-924f-c92dd553cca5.png)

**Comment by jonfroehlich** [comment]
As this problem appears to be on the user profile page, I'm closing this ticket.

The user profile map bug ticket owns this problem now: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/238

**Comment by jonfroehlich** [comment]
Ah oh, this is more serious than we thought. I just started auditing again. When I logged in, Fort Lincoln was shown as 0% complete (but I had forgotten that I had, in fact, actually audited it before). Then, once I started walking around, I'm like, gee this stuff seems very familiar and then I ended up being forced to to re-audit the paths that I already audited. Unfortunately, I didn't capture the intermediate missions screens (I had passed a few missions) but hopefully the screenshots below are enough to convince you that something is wrong.

This is a Mission Complete screenshot from 5 days ago (shown in https://github.com/ProjectSidewalk/SidewalkWebpage/issues/265#issue-173610973):
![image](https://cloud.githubusercontent.com/assets/1621749/18189840/28b027fc-708c-11e6-9d5f-7f05a5cc5111.png)

This is my current User Profile screen for Fort Lincoln. Notice how the routes that I audited before are not showing:
![image](https://cloud.githubusercontent.com/assets/1621749/18189820/176ef400-708c-11e6-8123-203ccd5a2323.png)

**Comment by kotarohara** [comment]
It seems like the data is there. So the problem is that the completed tasks are not tied correctly to Jon's account. This could be because the session timed out before you worked on the tasks and the data was not tied to your account. Either way, I'll take a look.

**Comment by jonfroehlich** [comment]
I just logged in and was assigned a 0.5 mile audit mission in Fort Lincoln. I just completed it. Here's what the Mission Complete screen looks like:

![image](https://cloud.githubusercontent.com/assets/1621749/18225545/78712ec6-71c2-11e6-8b17-c9ca01f997bd.png)

**Comment by jonfroehlich** [comment]
Now I am being re-routed through an audit area I've audited like three times before. I hope we can resolve this bug soon or I'll never be able to finish Fort Lincoln.

I just took the following screenshot:
![image](https://cloud.githubusercontent.com/assets/1621749/18225637/62c74d46-71c5-11e6-8618-1d37d9dcd8e9.png)

Does this screenshot look familiar? It should, because I already audited this area and took a screenshot (from the previous audit) and it's part of https://github.com/ProjectSidewalk/SidewalkWebpage/issues/263
